packages/datadog-plugin-fastify/src/fastify.js

```diff
@@ -77,13 +77,23 @@ function wrapMethod (method) {
   return function methodWithTrace (url, opts, handler) {
     const lastIndex = arguments.length - 1
 
-    arguments[lastIndex] = wrapHandler(arguments[lastIndex])
+    handler = arguments[lastIndex]
+
+    if (typeof handler === 'function') {
+      arguments[lastIndex] = wrapHandler(handler)
+    } else if (handler) {
+      arguments[lastIndex].handler = wrapHandler(handler.handler)
+    }
 
     return method.apply(this, arguments)
   }
 }
 
 function wrapHandler (handler) {
+  if (!handler || typeof handler !== 'function' || handler.name === 'handlerWithTrace') {
```

Comment:
It looks like there's no test covering `|| handler.name === 'handlerWithTrace'`. Should there be one here?

Reply:
This is just an optimization as it's always been double-patched but did not cause any issue. Basically, the behaviour doesn't change, but there is no point in double-patching.